### PR TITLE
SDL auxiliary library fixes

### PR DESCRIPTION
**Pull request by carstene1ns**
- This fixes freetype2 config file generation and use
- Installs needed SDL2 dependencies

With this and installed pkg-config all libraries build again on docker and macOS.

### CODE REVIEW

**Comment by sharkwouter** [comment]
Great, I'll try it

**Comment by sharkwouter** [comment]
This fixed SDL_ttf and Opentri not building on my machine

File: scripts/SDL2_gfx.sh [hunk 1]
```diff
@@ -1,4 +1,8 @@
 test_deps_install SDL2
 get_pspport SDL_gfx SDL2_gfx-psp
+
+# This is used when there is no pkg-config, but needs sdl2-config
+#EXTRA_FLAGS="--with-sdl-prefix=$(psp-config --psp-prefix)"
```
Comment (sharkwouter) [inline scripts/SDL2_gfx.sh:5]: Should this be commented out? I see this in SDL2_image and SDL2_mixer as well.

Comment (carstene1ns) [reply scripts/SDL2_gfx.sh:5]: Yes, this is intentionally disabled until we ship `sdl2.pc` or `sdl2-config`.
When shipping the pkg-config file, we do not need to bend `SDL2_CFLAGS`/`LIBS` anymore, when only shipping the config script, this needs to be enabled.
`AM_PATH_SDL2` will use whatever available, with preference to pkg-config nowadays.

Comment (sharkwouter) [reply scripts/SDL2_gfx.sh:5]: Makes sense, thanks!

**Pull request merged**